src/aui/dockart.cpp

```diff
@@ -685,7 +685,7 @@ wxAuiDefaultDockArt::DrawIcon(wxDC& dc, wxWindow *window, const wxRect& rect, wx
     }
 
     // Ensure the icon fits into the title bar.
-    wxSize iconSize = pane.icon.GetPreferredSizeFor(window);
+    wxSize iconSize = pane.icon.GetPreferredBitmapSizeFor(window);
     if (iconSize.y > rect.height)
```

Comment:
We can't compare `iconSize` in physical coordinates with `rect.height` which is in logical ones, so we need to convert one of them to the units of the other, e.g.
```suggestion
    if (iconSize.y > window->ToPhys(rect.height))
```